src/MobSpawner.cpp

```diff
@@ -107,14 +107,12 @@ eMonsterType cMobSpawner::ChooseMobType(EMCSBiome a_Biome)
 		}
 	}
 
+	// Pick a random mob from the options:
 	size_t allowedMobsSize = allowedMobs.size();
 	if (allowedMobsSize > 0)
 	{
-		std::set<eMonsterType>::iterator itr = allowedMobs.begin();
-
-		using DiffType = decltype(itr)::difference_type;
-		std::advance(itr, GetRandomProvider().RandInt<DiffType>(allowedMobsSize - 1));
-
+		auto itr = allowedMobs.begin();
+		std::advance(itr, GetRandomProvider().RandInt(allowedMobsSize - 1));
```

Comment:
Right, and using `difference_type` in MSVC generates an Intellisense parse error and compile-time warning. Stupid `std::advance`. Might as well rewrite it using a simple loop with none of the problems.